ts-impl.cc

```diff
@@ -128,6 +128,23 @@ struct LessThanAfter{
   }
 };
 
+struct StartsWith {
+  const char * const p;
+  const size_t s;
+  const size_t o;
+  StartsWith(const char * const p, const size_t s, const size_t o) :
+    p(p), s(s), o(o) {
+    ASSERT(p != NULL);
+    ASSERT(s <= strlen(p));
+  }
+  template < class I >
+  bool operator () (const I & i) const {
+    const char * const begin = i->pointer,
```

Comment:
We actually can't because we don't know the size of the input that we don't control (comes into the request). Example: we are looking starting on 2nd char on an empty string.